packages/eslint-plugin/src/configs/base-legacy.js

```diff
@@ -0,0 +1,18 @@
+// @ts-check
+
+const path = require('path');
+
+const { configHelpers } = require('..');
+
+/** @type {import("eslint").Linter.Config} */
+module.exports = {
+  extends: [path.join(__dirname, 'core')],
+  rules: {
+    /**
+     * `@typescript-eslint`plugin eslint rules
+     * @see https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin
+     */
+    ...configHelpers.getNamingConventionRule({ prefixInterface: true }),
```

Comment:
refactored the helper function to use named arguments instead of following boolean flags positional arguments which is a known antipattern (less readability/context, closed to extension etc....)